db/migrate/20180410092453_policy_log.rb

```diff
@@ -0,0 +1,87 @@
+Sequel.migration do
+  tables = %i(roles role_memberships resources permissions annotations)
+
+  up do
+    execute """
+      CREATE TYPE policy_log_op AS ENUM ('INSERT', 'DELETE', 'UPDATE');
+      CREATE TYPE policy_log_kind AS ENUM #{literal tables.map(&:to_s)};
+      CREATE EXTENSION IF NOT EXISTS hstore;
+    """
+
+    key = %i(policy_id version)
+    create_table :policy_log do
+      String :policy_id, null: false
+      Integer :version, null: false
+      foreign_key key, :policy_versions, on_delete: :cascade
+      index key
+
+      column :operation, :policy_log_op, null: false
+      column :kind, :policy_log_kind, null: false
+      column :subject, :hstore, null: false
+      column :at, :timestamptz, null: false, default: Sequel.function(:clock_timestamp)
```

Comment:
We will add `roles` and `resources` for indexing and query purposes?

Reply:
No, not here. This is just for pushing data out. A separate service will do querying.

Reply:
But it will want to see indexed columns on the table, right?

Reply:
It won't use this table at all :) I suppose you could think of that service as being an audit receiver, to (ab)use v4 parlance.